### DISCUSSION TITLE
Extend the json-schema of the Package

### DISCUSSION THREAD

**Discussion by engahmadkadi** [ENDER-9/Vocascan#106] [Ideas]
Hi Vocascan,

to bring the feature in #105 (use vocascan to learn proper sentences in a language and avoid common mistakes), is it possible to make the words in the sentence of first side of the tickets selectable? So the user would select the words that he thought are wrong or none proper, and once the ticket flips, depending on the selections, the tickets could evaluate to correct or wrong.

This might require expanding the format and fields of json-file, so an extra field should contain the index of wrong words in the expression.

Thanks and Regards

**Comment by luwol03** [comment]
Hey engahmadkadi,

as you might already have read in the last discussions, **I am** in favor of bringing vocascan to many other types of learning.

I can think of different kinds of interactive learning. Here are just a few ones:
- Cloze texts
- connecting boxes of information
- Covered areas of images
- selection lists
- ...

If you have more feel free to share them with us. 

However this will require a bit of refactor and introduction of different types of cards. 

But first of all I can think of adding some formatting to the card contents like markdown to be able to highlight some parts of a word/sentence.

luwol03

**Comment by engahmadkadi** [comment]
The following idea just came to my mind and would like to share it and write it down here.

Having my example of using Vocascan to learn the common mistakes in a language, I thought of having an option to switch to a different mode, not only the ticket flip mode, where:

- both side of the tickets will be shown at the same time.
- front- and backside have to shown in a random way, either right or links.
- the description of the backside should be hidden at this stage, as it might/should contains explanation, why this side is right and the other is wrong.
- both shown tickets must be clickable, as they the user should click what he thinks is the right sentence or word.
- by this click, the tool should know without asking the user for correct and wrong the state of the answer
- after the click, a highlight will show which one is right and which is wrong, meanwhile the description should appear to the user.

Ahmad

**Reply by engahmadkadi** [reply]
It seems this isn't the right place under this title, an if it worths creating an issue for it.
